naxsi_src/naxsi_utils.c

```diff
@@ -648,7 +648,11 @@ ngx_http_wlr_finalize_hashtables(ngx_conf_t *cf, ngx_http_dummy_loc_conf_t  *dlc
   }
 
   FILE *file;
-  char *fname = "/etc/nginx/whitelist.txt";
+  char fname[100];
+  if(dlc->whitelist_file)
+  {
+    memcpy(fname,dlc->whitelist_file->data,dlc->whitelist_file->len);
```

Comment:
please add a check to be sure that dlc->whitelist_file->len is not longer than the size of fname

Reply:
Done